File: Data-Structures-and-Algorithms/zerofilledarray/zerofilledarray.js
var zeroFilledSubarray = function(nums) {
    if (nums.length == 0) {
        return 0
    }
    
    let result = 0;
    let j = 0;
    for (var i=0; i < nums.length; i++) {
        if (nums[i] == 0) {
            result += i - j + 1
        };
        if (nums[i] != 0) {
            j = i + 1
        };
    }
    return result
};

arr = [0,0,0,2,0,0,2,0,0,0]
console.log(zeroFilledSubarray(arr))

// O(N) for time (N steps to iterate through)
// O(N) for space (constant extra space)

File: Data-Structures-and-Algorithms/zerofilledarray/zerofilledarray.py
class Functions(object):
    def zeroFilledSubarray(self, nums):
        total = 0
        cur = 0
        
        if len(nums) == 0:
            return 0
        
        for i in range(len(nums)):
            
            if nums[i] == 0:
                cur += 1
                
            if nums[i] != 0 or i == (len(nums)-1):
                if cur != 0:
                    total += ((cur+1)*cur/2)
                    cur = 0             
        return total
arr = [0,0,0,2,0,0,2,0,0]
print(arr)
print(Functions().zeroFilledSubarray(arr))

# O(N) for time (N steps to iterate through)
# O(1) for space (constant extra space)
